src/components/userGroups/K8sObjectPermissions/K8sListItemCard.tsx

```diff
@@ -0,0 +1,453 @@
+                ?.get?.(k8sPermission.group.value === 'k8sempty' ? '' : k8sPermission.group.value)
+                .child?.find((ele) => ele.gvk['Kind'] === selected.value)
+            const resourceListPayload: ResourceListPayloadType = {
+                clusterId: Number(k8sPermission?.cluster?.value),
+                k8sRequest: {
+                    resourceIdentifier: {
+                        groupVersionKind: resource.gvk,
+                    },
+                },
+            }
+            const { result } = await getResourceList(resourceListPayload)
+            setObjectMapping((prevMapping) => ({
+                ...prevMapping,
+                [k8sPermission.key]: [
+                    { label: 'All object', value: '*' },
+                    ...result.data.map((ele) => {
+                        return { label: ele['name'], value: ele['name'] }
+                    }),
+                ],
+            }))
+        } catch (err) {
+            showError(err)
+        }
+    }
+
+    const onClusterChange = (selected) => {
+        if (selected.value !== k8sPermission?.cluster?.value) {
+            handleK8sPermission('onClusterChange', index, selected)
+            getNamespaceList(selected.value)
+        }
+    }
+
+    const onNameSpaceSelection = (selected) => {
+        if (selected.value !== k8sPermission.namespace?.value) {
+            handleK8sPermission('onNamespaceChange', index, selected)
+            getGroupKindData(k8sPermission?.cluster?.value)
+        }
+    }
+
+    const onApiGroupSelect = (selected) => {
+        if (selected.value !== k8sPermission.group?.value) {
+            handleK8sPermission('onApiGroupChange', index, selected)
+            createKindData(selected)
+        }
+    }
+
+    const onKindSelect = (selected) => {
+        if (selected.value !== k8sPermission.kind?.value) {
+            handleK8sPermission('onKindChange', index, selected)
+            if (selected.value !== '*' && k8sPermission.group.value !== '*' && selected.value !== 'Event') {
+                getResourceListData(selected)
+            } else {
+                setObjectMapping((prevMapping) => ({
+                    ...prevMapping,
+                    [k8sPermission.key]: [{ label: 'All object', value: '*' }],
+                }))
+            }
+        }
+    }
+
+    const onObjectChange = (selected, actionMeta) => {
+        selectAllfunction(
+            selected,
+            actionMeta,
+            (options) => handleK8sPermission('onObjectChange', index, options),
+            objectMapping?.[k8sPermission.key],
+        )
+    }
+
+    const setRoleSelection = (selected) => {
+        if (selected.value !== k8sPermission.action?.value) {
+            handleK8sPermission('onRoleChange', index, selected)
+        }
+    }
+
+    const clonePermission = () => {
+        handleK8sPermission('clone', index)
+    }
+
+    const deletePermission = () => {
+        handleK8sPermission('delete', index)
+    }
```

Comment:
Merge both `clonePermission` & `deletePermission` and use dataset attr

Reply:
done